polynote-spark/src/main/scala/polynote/kernel/SparkPolyKernel.scala

```diff
@@ -150,6 +151,8 @@ class SparkPolyKernel(
     }
   }
 
+  override def cancelTasks(): IO[Unit] = super.cancelTasks() *> IO(DAGSchedulerThief(session).cancelAllJobs())
```

Comment:
Sometimes interrupting the thread doesn't stop the Spark tasks. This makes it work better (although also might kinda mess with the task progress notification... TODO)